src/public/assets/Dropdown.js

```diff
@@ -0,0 +1,147 @@
+const DATABASE = require('../../database.json');
+
+let nextID = 0;
+const DROPDOWNS = [];
+class Dropdown {
+  /**
+   * Creates a dropdown button
+   * @param {String} name The shortname of the button
+   * @param {String} text The button text
+   * @param {HTMLElement} parent The html parent of the button
+   * @param {Boolean} containsList If this dropdown has a list of sensors
+   * @param {Function} listOnClick The function the list element will execute on click
+   */
+  constructor(name, text, parent, containsList, listOnClick, chartIndex) {
```

Comment:
It could reduce the number of args you pass in here if you determine if there is a list just based on whether the "listOnClick" function is undefined or not. Just a possible simplification